api/bots/github_detail/readme.md

```diff
@@ -0,0 +1,28 @@
+# GitHub detail bot
+This bot links and details issues and pull requests.
+To use it username mention the bot then type an id:
+Ids can be specified in three different forms:
+- Id only: `#2000`
+- Repository and id: `zulip#2000`
+- Owner, repository and id `zulip/zulip#2000`
+
+Both the username mention and the id can occur at any time in the message. You
+can also mention multiple ids in a single message.
+
+The bot *requires* a default owner and repository to be configured.
```

Comment:
I wouldn't set this as a requisite if it allows doing searches in any repo (e.g. `zulip/zulip#2000`).

If you didn't specify some field in the config file (or there isn't any config file at all), and the user doesn't specify it in the query, that's when the bot should complain, IMO.